lib/common/api/deprecate.js

```diff
@@ -67,6 +67,25 @@ const deprecate = {
     })
   },
 
+  promisify: (promise, cb) => {
+    const oldName = `${promise.name} with callbacks`
+    const newName = `${promise.name} with Promises`
+    const warn = warnOnce(oldName, newName)
+
+    if (typeof cb !== 'function') return promise
+    return promise
+      .then(res => {
+        process.nextTick(() => {
+          warn()
```

Comment:
We should `warn` regardless of success or failure IMO.  I.e. This should probably be just above `return promise` on line 76